### PR TITLE
Improve GitHub contributing resources

### PR DESCRIPTION
**Pull request by kne42**
Improve various contributing resources including:
- [README](https://github.com/kne42/napari-gui/blob/gh-logistics/README.md)
- [contributing guidelines](https://github.com/kne42/napari-gui/blob/gh-logistics/CONTRIBUTING.md)
- [pull request template](https://github.com/kne42/napari-gui/blob/gh-logistics/.github/PULL_REQUEST_TEMPLATE.md?raw=True)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+$ git status
+```
+
+Add and commit your changed files:
+```sh
+$ git add my-file-or-directory
+$ git commit -m "my message"
+```
+
+> ### Help us make sure it's you
```
Comment (sofroniewn) [inline CONTRIBUTING.md]: Should this really be indented like a quote?

**Review by sofroniewn — APPROVED** [review]
Flagged one formatting thing, but otherwise looks good to me, nice job!

File: CONTRIBUTING.md (hunk 1 above)
Comment (kne42) [reply CONTRIBUTING.md]: TBH I was envisioning this as a “useful aside.” Not sure if a block quote properly captures this or something else would work better.

Comment (sofroniewn) [reply CONTRIBUTING.md]: Hmm, yeah - i found it a little confusing - not sure I see a problem with it being top level like the other sections, or maybe one down from `Making changes`

**Review by AhmetCanSolak — APPROVED** [review]
Thank you @kne42

File: CONTRIBUTING.md (hunk 1 above)
Comment (jni) [reply CONTRIBUTING.md]: I'm happy unindenting this, but I'm going to approve the PR in the meantime, good stuff. =)

**Review by jni — APPROVED** [review]
:+1:

**Pull request merged**